stdlib/Test/test/runtests.jl

```diff
@@ -722,6 +722,111 @@ end
     rm(f; force=true)
 end
 
+@testset "provide informative location in backtrace for test failures" begin
+    utils = tempname()
+    write(utils,
+    """
+    function test_properties2(value)
+        @test isodd(value)
+    end
+    """)
+
+    included = tempname()
+    write(included,
+    """
+    @testset "Other tests" begin
+        @test 1 + 1 == 3
+        test_properties2(2)
+    end
+    test_properties2(8)
```

Comment:
Maybe add a comment about the purpose of these? (Presumably these are designed to test cases where file/line info is missing?)